### PR TITLE
Adding inode availability in eviction policy

### PR DESCRIPTION
**Pull request by ronnielai**
Updates the proposal to include inodes as an eviction parameter.

### CODE REVIEW

File: docs/proposals/kubelet-eviction.md [hunk 1]
```diff
@@ -70,7 +70,9 @@ The `kubelet` will support the ability to trigger eviction decisions on the foll
 |------------------|---------------------------------------------------------------------------------|
 | memory.available | memory.available := node.status.capacity[memory] - node.stats.memory.workingSet |
 | nodefs.available   | nodefs.available := node.stats.fs.available |
+| nodefs.inodesFree | The available inodes under nodefs |
```
Comment (derekwaynecarr) [inline docs/proposals/kubelet-eviction.md]: do we plan on reporting this in summary API via something like node.fs and node.runtime.imageFs?

**Comment by derekwaynecarr** [comment]
I am +1 on the feature as it represents something our operators have hit in production.

File: docs/proposals/kubelet-eviction.md (hunk 1 above)
Comment (ronnielai) [inline docs/proposals/kubelet-eviction.md]: I think we should. I've updated the doc to reflect that. Thx.

File: docs/proposals/kubelet-eviction.md [hunk 2]
```diff
@@ -70,7 +70,9 @@ The `kubelet` will support the ability to trigger eviction decisions on the foll
 |------------------|---------------------------------------------------------------------------------|
 | memory.available | memory.available := node.status.capacity[memory] - node.stats.memory.workingSet |
 | nodefs.available   | nodefs.available := node.stats.fs.available |
+| nodefs.inodesFree | nodefs.inodesFree := node.fs.inodeFree |
 | imagefs.available | imagefs.available := node.stats.runtime.imagefs.available |
+| imagefs.inodesFree | imagefs.inodesFree := node.runetime.imageFs.inodeFree |
```
Comment (vishh) [inline docs/proposals/kubelet-eviction.md]: nit: `s/runetime/runtime`

**Comment by vishh** [comment]
@derekwaynecarr Any inputs on what should be sane defaults for minimum number of inodes?

**Comment by vishh** [comment]
Adding this feature LGTM

File: docs/proposals/kubelet-eviction.md (hunk 2 above)
Comment (ronnielai) [inline docs/proposals/kubelet-eviction.md]: fixed.

**Comment by derekwaynecarr** [comment]
As for sane amounts I know our operations team monitors this now.  I will
follow up after the 4th of July holiday.

**Comment by derekwaynecarr** [comment]
After chatting with our ops team, and exploring the other solutions in this space, there is a request that we support expressing this eviction in terms of percentage of inodes available.  So we would do a soft threshold when inode available falls below 10% and a hard threshold when it falls below 5%..  Unlike imagefs where using percentages could result in a lot of waste with large file systems, it appears with inodes percentages are preferred.

**Comment by derekwaynecarr** [comment]
@vishh @ronnielai can we update the eviction signal to be something that represents a percentage of inodes available before we get too far in implementation?  I see there were a number of PRs in flight.

**Pull request merged**